platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -760,7 +933,57 @@ namespace platoon_strategic_ihp
+                 * UCLA Implementation note:
+                 *  1. isVehicleNextLane --> determine if the platoon is next to host vheicle
+                 *  2. sender_id == requesting veh ID --> front joiner ID
+                 */
+                std::string newLeaderPlatoonId = pm_.currentPlatoonID; // set self's (front join veh) platoon ID as the new platoon ID
+                ROS_DEBUG_STREAM("Host vehicle platoon id = " << newLeaderPlatoonId << "; join the rear platoon as the new leader (frontal join).");
+                
+                // compose request
+                cav_msgs::MobilityRequest request;
+                request.header.plan_id = boost::uuids::to_string(boost::uuids::random_generator()());
+                request.header.recipient_id = senderId;
+                request.header.sender_id = config_.vehicleID;
+                request.header.timestamp = ros::Time::now().toNSec() / 1000000;
+                request.location = pose_to_ecef(pose_msg_);
+                // UCLA: Desired joining index for cut-in join, indicate the index of gap-leading vehicle. -1 indicate cut-in from front.
+                request.join_index = -1; // Note: not used by same-lane functions.
+                
+                // UCLA: change plan type --> ADD THIS TO PLANTYPE.MSG !!
+                request.plan_type.type = cav_msgs::PlanType::CUT_IN_FROM_SIDE; // UCLA: --> add this in plantype.msg
+                request.strategy = MOBILITY_STRATEGY;
+                
+                // get platoon size
+                int platoon_size = pm_.getTotalPlatooningSize();
+
+                /**
+                 * JOIN_CUT_IN_PARAMS format: 
+                 *       JOIN_CUT_IN_PARAMS| --> "SIZE:%1%,SPEED:%2%,ECEFX:%3%,ECEFY:%4%,ECEFZ:%5%"
+                 *                            |-------0------ --1---------2---------3---------4------|
+                 */
+                boost::format fmter(JOIN_CUT_IN_PARAMS); // Note: Front and rear join uses same params, hence merge to one param for both condition.
+                fmter %platoon_size;                //  index = 0
+                fmter %current_speed_;              //  index = 1, in m/s
+                fmter %pose_ecef_point_.ecef_x;     //  index = 2
+                fmter %pose_ecef_point_.ecef_y;     //  index = 3
+                fmter %pose_ecef_point_.ecef_z;     //  index = 4
+
+                request.strategy_params = fmter.str();
+                request.urgency = 50;
+
+                // UCLA: generate new platoonplan for front join.
+                pm_.current_plan = PlatoonPlan(true, request.header.timestamp, request.header.plan_id, request.header.sender_id);
+                mobility_request_publisher_(request);
+                ROS_DEBUG_STREAM("Publishing request to the leader " << senderId << " with params " << request.strategy_params << " and plan id = " << request.header.plan_id);
+                potentialNewPlatoonId_front = newLeaderPlatoonId;
```

Comment:
Would be nice if potentialNewPlatoonId_front would use consistent style: either camel case or snake case throughout, and end with a _ to indicate it is a member variable.